packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterAuthRule.kt

```diff
@@ -16,10 +16,74 @@
 package com.amazonaws.amplify.amplify_datastore.types.model
 
 import com.amplifyframework.core.model.AuthRule
+import com.amplifyframework.core.model.AuthStrategy
+import com.amplifyframework.core.model.ModelField
+import com.amplifyframework.core.model.ModelOperation
 
 data class FlutterAuthRule(val map: Map<String, Any>) {
+    
+    private val authStrategy : AuthStrategy =
+            indexToAuthStrategy(map["authStrategy"] as Int)
+    private val ownerField : String? = map["ownerField"] as String?
+    private val identityClaim : String? = map["identityClaim"] as String?
+    private val groupClaim : String? = map["groupClaim"] as String?
+    private val groups : List<String> = map["groups"] as List<String>
```

Comment:
You are sure that `groups` doesn't need the optional `?`

Reply:
You're right it can be optional as well, added the ? to groups